test/lib/custom-assertions.js

```diff
@@ -80,6 +80,95 @@ function compareSegments(parent, segments) {
   })
 }
 
+/**
+ * @param {TraceSegment} parent     Parent segment
+ * @param {Array} expected          Array of strings that represent segment names.
+ *                                  If an item in the array is another array, it
+ *                                  represents children of the previous item.
+ * @param {boolean} options.exact   If true, then the expected segments must match
+ *                                  exactly, including their position and children on all
+ *                                  levels.  When false, then only check that each child
+ *                                  exists.
+ * @param {array} options.exclude   Array of segment names that should be excluded from
+ *                                  validation.  This is useful, for example, when a
+ *                                  segment may or may not be created by code that is not
+ *                                  directly under test.  Only used when `exact` is true.
+ */
+function assertSegments(parent, expected, options) {
+  let child
+  let childCount = 0
+
+  // rather default to what is more likely to fail than have a false test
+  let exact = true
+  if (options && options.exact === false) {
+    exact = options.exact
+  } else if (options === false) {
+    exact = false
+  }
+
+  function getChildren(_parent) {
+    return _parent.children.filter(function (item) {
+      if (exact && options && options.exclude) {
+        return options.exclude.indexOf(item.name) === -1
+      }
+      return true
+    })
+  }
+
+  const children = getChildren(parent)
+  if (exact) {
+    for (let i = 0; i < expected.length; ++i) {
+      const sequenceItem = expected[i]
+
+      if (typeof sequenceItem === 'string') {
+        child = children[childCount++]
+        assert.equal(
+          child ? child.name : undefined,
+          sequenceItem,
+          'segment "' +
+            parent.name +
+            '" should have child "' +
+            sequenceItem +
+            '" in position ' +
+            childCount
+        )
+
+        // If the next expected item is not array, then check that the current
+        // child has no children
+        if (!Array.isArray(expected[i + 1])) {
+          // var children = child.children
```

Comment:
Should this commented-out line be removed?

Reply:
i can remove. i copied this function and made it non tap assertions.